drafts/mvvm_toolkit.md

```diff
@@ -0,0 +1,260 @@
+    }
+
+    private void ResetNickname() => Nickname = string.Empty;
+}
+```
+
+   Describes how the data between the view and the model flows. Covers the MV characters in MVVM. In this file, I would like to emphasize some code pieces namely INotifyPropertyChanged, ResetNicknameCommand, the set block of the Nickname property, PropertyChanged and OnPropertyChanged.
+
+#### INotifyPropertyChanged
+
+   This interface is needed to tell .NET MAUI that the implementing class has properties that can change and can trigger the rerendering of the view.
+
+#### ResetNicknameCommand
+
+   This property encapsulates a behavior that can be triggered from interactable UI elements. In this case, it invokes the method ResetNickname.
+
+#### Set block of Nickname property
+
+   Checks if the new value is different than the stored value. This check prevents unnecessary re-renders. Afterward sets the new value and invokes OnPropertyChanged.
+
+#### PropertyChanged
+
+   Is an event that notifies .NET MAUI that the calling property has changed and that all bound UI elements need to be re-rendered.
+
+#### OnPropertyChanged
+
+   The method that invokes the event PropertyChanged. Gets the name of the changed property via the attribute CallerMemberName.
+
+1. Cat.cs
+
+   Describes the data to display.
+
+```C#
+namespace CatFinder.Model;
+
+public record Cat(string Name, string ImageUrl, string Temperament, string CountryCodes, int Intelligence)
+{
+    public string Nickname { get; set; }
+}
+```
+
+Covers the M character in MVVM. In this example, I used the record keyword to define the type. The record type is a feature that was released with C#9. A record brings a lot of benefits with it, like an override of Object.Equals(Object), methods for operator == and operator !=,  primary constructors, etc. These features make a record the perfect candidate for plain data objects.
+
+### Introduction to the MVVM Community Toolkit
+
+This chapter gives an overview of the MVVM Community Toolkit and shows how the cat details page from the previous chapter can be implemented with the MVVM Community Toolkit.
+
+The [MVVM Community Toolkit](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm) is a well-maintained and mature library developed by Microsoft and the .NET foundation.
+
+The MVVM Community Toolkit can be installed via the NuGet installer in Visual Studio. It allows us developers to reduce the amount of code we need to write to implement MVVM in XAML-based solutions. The concept used to reduce the boiler code by the MVVM Community Toolkit is source generators. In short, using elements of the MVVM Community Toolkit in our code leads to generated code that is outsourced in different files. These generated files implement exactly the verbose behavior that we would need to write without the MVVM Community Toolkit. I will not dig deeper into this topic because it is a highly interesting topic for a future blog post.
```

Comment:
```suggestion
The MVVM Community Toolkit can be installed via the NuGet installer in Visual Studio. It allows us developers to reduce the amount of code we need to write to implement MVVM in XAML-based solutions. The concept used to reduce the boilerplate code by the MVVM Community Toolkit is source generators. In short, using elements of the MVVM Community Toolkit in our code leads to generated code that is outsourced in different files. These generated files implement exactly the verbose behavior that we would need to write without the MVVM Community Toolkit. I will not dig deeper into this topic because it is a highly interesting topic for a future blog post.
```

Reply:
This sentence: `I will not dig deeper into this topic because it is a highly interesting topic for a future blog post.` feels a little bit disappointing like this. Maybe point out, that the topic is a bit big for this blog post, that's why you'll cover it in a future blog post or something similar to that.

Reply:
You are completely right. I rewrote the sentence, making it more positive.